ckanext/followy/tests/e2e/followy_test.py

```diff
@@ -0,0 +1,110 @@
+        self.addCleanup(self.browser.quit)
+
+
+    def test_create_dataset_and_follow(self):
+        self.browser.get('http://localhost:5000')
+        login = self.browser.find_element_by_link_text("Log in")
+        login.click()
+        username = self.browser.find_element_by_id("field-login")
+        password = self.browser.find_element_by_id("field-password")
+        login_Btn = self.browser.find_elements_by_class_name("btn-primary")[0]
+        username.send_keys("ghost")
+        password.send_keys("ghostmode")
+        login_Btn.click()
+        assert 'http://localhost:5000/dashboard' in self.browser.current_url
+        datasets = self.browser.find_element_by_link_text("Datasets")
+        datasets.click()
+        assert 'http://localhost:5000/dataset' in self.browser.current_url
+        add_dataset = self.browser.find_element_by_link_text("Add Dataset")
+        search_dataset = self.browser.find_elements_by_class_name("search")
+        assert add_dataset != None 
+        assert search_dataset != None
+        add_dataset.click()
+        assert 'http://localhost:5000/dataset/new' in self.browser.current_url
+        title = self.browser.find_element_by_id("field-title")
+        assert title != None
+        description = self.browser.find_element_by_id("field-notes")
+        assert description != None
+        visibility = self.browser.find_element_by_id("field-private")
+        visibility != None
+        next_page = self.browser.find_elements_by_class_name("btn-primary")[0]
+        assert next_page != None
+        title_text = "".join( [random.choice(string.ascii_letters) for i in range(8)] )
+        title.send_keys(title_text)
+        description_text = "".join( [random.choice(string.ascii_letters) for i in range(20)] )
+        description.send_keys(description_text)
+        time.sleep(4)
+        next_page.click()
+        dataset_url = self.browser.find_element_by_name("url")
+        assert dataset_url != None
+        url_text = "".join( [random.choice(string.ascii_letters) for i in range(12)] )
+        dataset_url.send_keys(url_text)
+        finish_btn = self.browser.find_elements_by_class_name("btn-primary")[0]
+        assert finish_btn != None
+        finish_btn.click()
+        activity_stream = self.browser.find_elements_by_class_name("icon-time")
+        assert activity_stream != None
+        manage_btn = self.browser.find_elements_by_class_name("icon-wrench")
+        assert manage_btn != None
+        follow_btn = self.browser.find_elements_by_class_name("btn-success")[0]
+        assert follow_btn != None
+        follow_btn.click()
+        success_message = self.browser.find_elements_by_class_name("alert-success")
+        assert success_message != None
+        time.sleep(4)        
+
+
+    def test_following_dataset(self):
+        self.browser.get('http://localhost:5000')
+        login = self.browser.find_element_by_link_text("Log in")
```

Comment:
The code here #71 to #78 is repeated as it appears in a similar way in the test add dataset function above it, to me this kills DRY philosophy, Have you considered doing a helper function to handle the authentication or just fo it in the set up since both tests in the test class rely on Authentication

Reply:
noted @shemogumbe I would refactor it, using a helper function like you said.